lnrpc/routerrpc/router_backend.go

```diff
@@ -702,9 +702,31 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 			payIntent.MaxParts = 1
 		}
 
-		err = payIntent.SetPaymentHash(*payReq.PaymentHash)
-		if err != nil {
-			return nil, err
+		if payReq.Features.HasFeature(lnwire.AMPOptional) {
```

Comment:
See prior comment about also setting a random _pay addr_ here if the pay addr is blank, or an additional optional feature bit is present.